packages/app/cypress/e2e/index.cy.ts

```diff
@@ -1,181 +1,737 @@
 import defaultMessages from '@packages/frontend-shared/src/locales/en-US.json'
-import type { Interception } from '@packages/net-stubbing/lib/external-types'
-import type { FoundSpec } from '@packages/types/src'
 
 describe('App: Index', () => {
-  beforeEach(() => {
-    cy.scaffoldProject('non-existent-spec')
-    cy.openProject('non-existent-spec')
-    cy.withCtx(async (ctx, { testState }) => {
-      testState.newFilePath = 'cypress/e2e/new-file.cy.js'
+  describe('Testing Type: E2E', () => {
+    context('project with default spec pattern', () => {
+      beforeEach(() => {
+        cy.scaffoldProject('no-specs-no-storybook')
+        cy.openProject('no-specs-no-storybook')
+        cy.startAppServer('e2e')
+        cy.visitApp()
+
+        // With no specs present, the page renders two cards, one for scaffolding example specs,
+        // another for creating a new blank spec.
+        cy.findAllByTestId('card').eq(0).as('ScaffoldCard')
+        .should('have.attr', 'tabindex', '0')
+        .within(() => {
+          cy.findByRole('heading', {
+            level: 2, name: defaultMessages.createSpec.e2e.importFromScaffold.header,
+          }).should('be.visible')
+
+          cy.contains(defaultMessages.createSpec.e2e.importFromScaffold.description)
+          .should('be.visible')
+        })
+
+        cy.findAllByTestId('card').eq(1).as('EmptySpecCard')
+        .should('have.attr', 'tabindex', '0')
+        .within(() => {
+          cy.findByRole('heading', {
+            level: 2, name: defaultMessages.createSpec.e2e.importEmptySpec.header,
+          }).should('be.visible')
+
+          cy.contains(defaultMessages.createSpec.e2e.importEmptySpec.description)
+          .should('be.visible')
+        })
+      })
+
+      it('shows create first spec page with scaffold and create empty spec options', () => {
+        cy.findByRole('heading', {
+          level: 1,
+          name: defaultMessages.createSpec.page.defaultPatternNoSpecs.title,
+        }).should('be.visible')
+
+        cy.findByTestId('create-spec-page-description').should('be.visible')
+        .and('contain', defaultMessages.createSpec.page.defaultPatternNoSpecs.e2e.description)
+
+        cy.get('@ScaffoldCard').should('be.visible')
+        cy.get('@EmptySpecCard').should('be.visible')
+
+        cy.findByTestId('no-specs-message').should('be.visible')
+        .and('contain', defaultMessages.createSpec.noSpecsMessage)
+
+        cy.findByRole('link', { name: defaultMessages.createSpec.viewSpecPatternButton })
+        .should('be.visible')
+        .and('not.be.disabled')
+        .and('have.attr', 'href', '#/settings?section=project&setting=specPattern')
```

Comment:
is this href at risk of changing?

Reply:
Not in the short term. Medium term, this type of deeplink to the spec pattern will be replaced with a modal the opens and displays the spec pattern directly. But for now, the settings page is explicitly looking for this query param to decide where to scroll, so it's a pretty stable situation and we'd only want to change this URL if that implementation changed.

Reply:
@marktnoonan is there an issue logged to update to the modal pattern? I wasn't able to find one.

Reply:
@tbiethman there was an issue, but recently closed as we believe this change has been dropped from 10.0. But it is still in Figma and I expect it will come around soon enough even if not present during release. It's probably a nicer experience to just peek at the spec pattern instead of leaving the current page.

Reply:
@tbiethman I stand corrected, @ZachJW34 has this change completed as part of other spec pattern work [in this PR](https://github.com/cypress-io/cypress/pull/19801/), so whichever merges last can align the test with the new behavior.

Reply:
@marktnoonan I'll keep an eye on that PR. It doesn't change the specs page implementation, at least as it stands. But if the component work is done, it might make sense to re-open that issue and hook this up prior to release.